### DISCUSSION TITLE
Worker executions question

### DISCUSSION THREAD

**Discussion by CappiSteijns** [Zibbp/ganymede#475] [Q&A]
I am currently using the following parameters on my Unraid server:

```
# WORKER
- MAX_CHAT_DOWNLOAD_EXECUTIONS=5
- MAX_CHAT_RENDER_EXECUTIONS=3
- MAX_VIDEO_DOWNLOAD_EXECUTIONS=5
- MAX_VIDEO_CONVERT_EXECUTIONS=3
```

I’ve noticed that my server becomes very busy, especially CPU-wise, when there are 3 simultaneous render executions or 3 video convert executions.

I am considering changing the values for MAX_CHAT_RENDER_EXECUTIONS and MAX_VIDEO_CONVERT_EXECUTIONS from "3" to "1". My question is:
1: If I set these values to "1", will the tasks just be queued up and executed sequentially without causing performance issues?

Additionally, if there are more than 5 live channels being watched at the same time:

2: Will the channels beyond the limit of "5" simply not be recorded?
3: Would it be safer to increase the MAX_CHAT_DOWNLOAD_EXECUTIONS and MAX_VIDEO_DOWNLOAD_EXECUTIONS values to a higher number to ensure all channels are recorded?

Thank you in advance for your help.

**Comment by Zibbp** [comment]
> If I set these values to "1", will the tasks just be queued up and executed sequentially without causing performance issues?

Yes if all of these values are set to 1 then tasks will be queued up and worked sequentially.

> 2: Will the channels beyond the limit of "5" simply not be recorded?
3: Would it be safer to increase the MAX_CHAT_DOWNLOAD_EXECUTIONS and MAX_VIDEO_DOWNLOAD_EXECUTIONS values to a higher number to ensure all channels are recorded?

Live archive (video and chat downloads) bypass the execution limit restrictions. This means that they will always download the video and chats. Converting the video and rendering the chat *after* the live stream ends will be held if there aren't any free execution slots. So yes, It's fine to set these values to 1. It may get backed up though, if it can't keep up with the archive demand you may need to increase a specific variable.

These settings are more apparent in the [v3 beta](https://github.com/Zibbp/ganymede/issues/474). You will be able to see if a task is in a pending state due to not enough execution slots.
![image](https://github.com/user-attachments/assets/afe30c90-3202-4e00-b669-62a7bb2783ab)

**Reply by CappiSteijns** [reply]
Thank you for your quick response and for the continuous development of Ganymede.

Version 3 looks very promising with some great new features, especially the easy option to restart a workflow.

Thanks
